internal/manifests/imagestream/imagestream.go

```diff
@@ -0,0 +1,23 @@
+package imagestream
+
+import (
+	"context"
+
+	"github.com/ViaQ/logerr/v2/kverrors"
+	imagev1 "github.com/openshift/api/image/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Get returns the k8s imagestream for the given object key or an error.
+func Get(ctx context.Context, c client.Client, key client.ObjectKey) (*imagev1.ImageStream, error) {
+	is := New(key.Name, key.Namespace)
+
+	if err := c.Get(ctx, key, is); err != nil {
+		return is, kverrors.Wrap(err, "failed to get imagestream",
+			"name", is.Name,
+			"namespace", is.Namespace,
+		)
+	}
```

Comment:
I would like to keep this function separate so that in future if we ever need to add more functions related to imagestream then we don't have to refractor again. Also, it keeps our codebase consistent.

Reply:
One of my points was that the current `manifests/imagestream` is already not consistent with the other packages in that directory, because it is, as far as I can see, the only one where we only consume the resource and not write new resources based on the decisions taken in the operator.

The code in `New()` seems not to be needed for the `Get()`. I have tested with just passing an empty `*imagev1.ImageStream{}` and this seems to work fine as well. I think replacing these two files with ~40 lines of code with the seven lines that we actually need is worth it.

Unless we can already see now that we need to extend the ImageStream code in the near future, I think it is not necessary to keep the package just so that it is easily extendable in the future.